### DISCUSSION TITLE
Training a yoloV9 in kaggle using the kaggle 20 hour per week tpu and saving version in cloud

### DISCUSSION THREAD

**Discussion by lukorito99** [ultralytics/ultralytics#15731] [Q&A]
Requesting guide to training a yoloV9 in kaggle using the kaggle 20 hour per week tpu and succesfully saving version in cloud

**Comment by glenn-jocher** [comment]
@lukorito99 to train a YOLOv9 model on Kaggle using TPUs and save your work, please refer to our detailed guide at https://docs.ultralytics.com/integrations/kaggle/. This resource covers setting up TPUs, running the training, and saving your notebook versions.

**Comment by lukorito99** [comment]
This is my attempt but no success;
import tensorflow as tf
try:
    tpu = tf.distribute.cluster_resolver.TPUClusterResolver() 
    print('Running on TPU ', tpu.master())
except ValueError:
    tpu = None

if tpu:
    tf.config.experimental_connect_to_cluster(tpu)
    tf.tpu.experimental.initialize_tpu_system(tpu)
    strategy = tf.distribute.experimental.TPUStrategy(tpu)
else:
    strategy = tf.distribute.get_strategy() 

print("REPLICAS: ", strategy.num_replicas_in_sync)

from ultralytics import YOLO
with strategy.scope():
    model = YOLO("yolov10x.yaml")
    data_path = '/kaggle/input/balldetection/Bundesliga Balls.v5i.yolov9/data.yaml'

metrics = model.train(data=data_path, epochs=500, imgsz=1280)
metrics = model.val()
print(f'mAP: {metrics.box.map}, mAP@50: {metrics.box.map50}, mAP@75: {metrics.box.map75}, mAPs: {metrics.box.maps}')

# Export the model
model.export()
model.export(format="onnx")
model.export(format="onnx",int8 = True)
model.export(format="tflite")
model.export(format="tflite",int8 = True)

**Reply by glenn-jocher** [reply]
@lukorito99 it looks like you're trying to train a YOLOv10 model on Kaggle using TPUs. Please note that YOLOv10 is not supported by Ultralytics. Instead, use YOLOv8 or another supported model. Ensure you have the latest package version and follow the Kaggle integration guide: https://docs.ultralytics.com/integrations/kaggle/. If issues persist, please provide more details.
